scripts/NuGetUtilities.psm1

```diff
@@ -13,13 +13,16 @@ $InformationPreference = "Continue"
 Import-Module $PSScriptRoot\ScriptUtilities.psm1
 Import-Module $PSScriptRoot\Projects.psm1
 
-$NugetExePath = "$RepoRoot\.nuget\NuGet.exe"
-$NuGetPackageRoot = "$SourceRoot\packages"
-$NuGetSamplesPackageRoot = "$SourceRoot\Samples\packages"
-$NuGetConfigFile = "$RepoRoot\NuGet.Config"
+$NugetExePath = "nuget"
+if ($ENV:OS) {
+    $NugetExePath = "$RepoRoot\.nuget\NuGet.exe"
+}
+$NuGetPackageRoot = Join-Path $SourceRoot "packages"
+$NuGetSamplesPackageRoot = Join-Path (Join-Path $SourceRoot "Samples") "packages"
```

Comment:
>-Path (Join-Path $SourceRoot "Samples") "packages" [](http://example.com/codeflow?start=31&length=50)

can you check if we can simplify this by using:
[https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/join-path?view=powershell-7.2#example-7--combine-an-indefinite-number-of-paths](https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/join-path?view=powershell-7.2#example-7--combine-an-indefinite-number-of-paths)

Reply:
if yes, update all places where u have join-path (join-path...

Reply:
oh yes I begin with that simplified syntax, it works in my machine, but not in the build server, the PowerShell version is too low in build server.

Reply:
the task is named:
- task: PowerShell@2
the 2 can not be changed,
but it can add this line
pwsh: true
to change to use PowerShell Core!
the simple syntax works after the change.